tests/time_tests/scripts/run_timetest.py

```diff
@@ -95,6 +109,13 @@ def run_timetest(args: dict, log=None):
         stats = dict((step_name, stats.get(step_name, []) + [duration])
                      for step_name, duration in raw_data.items())
 
+    # Remove outliers
+    for step_name, time_results in stats.items():
+        iqr, q1, q3 = calculate_iqr(time_results)
```

Comment:
@just-sparta may me move all calculation and filtering to a utilities to decrease number fo changes in `run_timetest.py`?